File: extensions/instagram/src/main/java/app/argusos/extension/instagram/misc/share/privacy/SanitizeSharingLinksPatch.java
package app.argusos.extension.instagram.misc.share.privacy;

import app.argusos.extension.shared.privacy.LinkSanitizer;

@SuppressWarnings("unused")
public final class SanitizeSharingLinksPatch {
    private static final LinkSanitizer sanitizer = new LinkSanitizer("igsh");

    /**
     * Injection point.
     */
    public static String sanitizeSharingLink(String url) {
        return sanitizer.sanitizeURLString(url);
    }
}

File: patches/src/main/kotlin/app/argusos/patches/music/misc/privacy/SanitizeSharingLinksPatch.kt
package app.argusos.patches.music.misc.privacy

import app.argusos.patches.music.misc.extension.sharedExtensionPatch
import app.argusos.patches.music.misc.settings.PreferenceScreen
import app.argusos.patches.music.misc.settings.settingsPatch
import app.argusos.patches.shared.misc.privacy.sanitizeSharingLinksPatch

@Suppress("unused")
val sanitizeSharingLinksPatch = sanitizeSharingLinksPatch(
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch,
        )

        compatibleWith(
            "com.google.android.apps.youtube.music"(
                "7.29.52",
                "8.10.52",
                "8.37.56",
                "8.40.54"
            )
        )
    },
    preferenceScreen = PreferenceScreen.MISC,
    replaceMusicLinksWithYouTube = true
)